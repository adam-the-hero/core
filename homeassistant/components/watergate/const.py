"""Constants for the Watergate integration."""

DOMAIN = "watergate"

SONIC_NAME = "sonic_name"
SONIC_ADDRESS = "sonic_address"

MANUFACTURER = "Watergate AI"

COORDINATOR = "watergate-coordinator"
API = "api-client"

VALVE_SENSOR_NAME = "Valve"
WIFI_CONNECTION_SENSOR_NAME = "Wifi connection"
MQTT_CONNECTION_SENSOR_NAME = "MQTT connection"
UPTIME_SENSOR_NAME = "Uptime"
UPTIME_SENSOR_NAME = "Uptime"
POWER_SUPPLY_SENSOR_NAME = "Power supply"
WATER_FLOWING_SENSOR_NAME = "Water flowing"
PRESSURE_SENSOR_NAME = "Water pressure"
TEMPERATURE_SENSOR_NAME = "Water temperature"
FLOW_SENSOR_NAME = "Water flow rate"
IP_SENSOR_NAME = "IP address"
GATEWAY_SENSOR_NAME = "Gateway address"
SUBNET_SENSOR_NAME = "Subnet"
SSID_SENSOR_NAME = "SSID"
RSSI_SENSOR_NAME = "RSSI"
WIFI_UPTIME_SENSOR_NAME = "WIFI Uptime"
MQTT_UPTIME_SENSOR_NAME = "MQTT Uptime"
SHUT_OFF_SENSOR_NAME = "Auto Shut off"
SHUT_OFF_VOLUME_SENSOR_NAME = "Auto Shut off volume"
SHUT_OFF_DURATION_SENSOR_NAME = "Auto Shut off duration"
SHUT_OFF_EVENT_SENSOR_NAME = "Last Auto Shut off event"
WATER_METER_VOLUME_SENSOR_NAME = "Water meter volume"
WATER_METER_DURATION_SENSOR_NAME = "Water meter duration"


VALVE_ENTITY_NAME = "valve"
WIFI_CONNECTION_ENTITY_NAME = "wifi_status"
MQTT_CONNECTION_ENTITY_NAME = "mqtt_status"
UPTIME_ENTITY_NAME = "uptime"
POWER_SUPPLY_ENTITY_NAME = "power_supply"
WATER_FLOWING_ENTITY_NAME = "water_flowing"
PRESSURE_ENTITY_NAME = "water_pressure"
TEMPERATURE_ENTITY_NAME = "water_temperature"
FLOW_ENTITY_NAME = "water_flow_rate"
IP_ENTITY_NAME = "ip"
GATEWAY_ENTITY_NAME = "gateway"
SUBNET_ENTITY_NAME = "subnet"
SSID_ENTITY_NAME = "ssid"
RSSI_ENTITY_NAME = "rssi"
WIFI_UPTIME_ENTITY_NAME = "wifi_uptime"
MQTT_UPTIME_ENTITY_NAME = "mqtt_uptime"
SHUT_OFF_ENTITY_NAME = "auto_shut_off"
SHUT_OFF_VOLUME_ENTITY_NAME = "auto_shut_off_volume"
SHUT_OFF_DURATION_ENTITY_NAME = "auto_shut_off_duration"
SHUT_OFF_EVENT_ENTITY_NAME = "auto_shut_off_event"
WATER_METER_VOLUME_ENTITY_NAME = "water_meter_volume"
WATER_METER_DURATION_ENTITY_NAME = "water_meter_duration"

ASO_EVENT_TYPE_ATRIBUTE = "type"
ASO_EVENT_VOLUME_ATRIBUTE = "volume"
ASO_EVENT_DURATION_ATRIBUTE = "duration"

WEBHOOK_ASO_REPORT_TYPE = "auto-shut-off-report"
WEBHOOK_TELEMETRY_TYPE = "telemetry"
WEBHOOK_VALVE_TYPE = "valve"
WEBHOOK_WIFI_CHANGED_TYPE = "wifi-changed"
WEBHOOK_POWER_SUPPLY_CHANGED_TYPE = "power-supply-changed"
