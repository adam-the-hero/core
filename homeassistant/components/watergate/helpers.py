"""Common functions used in watergate integration."""

from homeassistant.config_entries import ConfigEntry
from homeassistant.helpers.device_registry import DeviceInfo

from .const import DOMAIN, MANUFACTURER, SONIC_NAME
from .coordinator import WatergateDataCoordinator


def extract_firmware_version(coordinator: WatergateDataCoordinator) -> str:
    """Extract firmware version from coordinator."""
    return coordinator.data.state.firmware_version if coordinator.data.state else ""


def get_device_info(
    coordinator: WatergateDataCoordinator, entry: ConfigEntry
) -> DeviceInfo:
    """Return device info for a Watergate device."""

    return DeviceInfo(
        identifiers={(DOMAIN, entry.entry_id)},
        name=entry.data[SONIC_NAME],
        manufacturer=MANUFACTURER,
        sw_version=extract_firmware_version(coordinator),
    )
