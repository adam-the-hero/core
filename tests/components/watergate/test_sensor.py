"""Tests for the Watergate valve platform."""

from collections.abc import Generator

import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.const import EntityCategory, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import init_integration
from .const import MOCK_WEBHOOK_ID

from tests.common import AsyncMock, MockConfigEntry, patch, snapshot_platform
from tests.typing import ClientSessionGenerator


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_sensor(
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
    mock_entry: MockConfigEntry,
    mock_watergate_client: Generator[AsyncMock],
    snapshot: SnapshotAssertion,
) -> None:
    """Test states of the sensor."""
    with patch("homeassistant.components.watergate.PLATFORMS", [Platform.SENSOR]):
        await init_integration(hass, mock_entry)

    await snapshot_platform(hass, entity_registry, snapshot, mock_entry.entry_id)


async def test_diagnostics_are_disabled_by_default(
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
    mock_entry: MockConfigEntry,
    mock_watergate_client: Generator[AsyncMock],
) -> None:
    """Test if all diagnostic entities are disabled by default."""
    with patch("homeassistant.components.watergate.PLATFORMS", [Platform.SENSOR]):
        await init_integration(hass, mock_entry)

    entries = [
        entry
        for entry in entity_registry.entities.get_entries_for_config_entry_id(
            mock_entry.entry_id
        )
        if entry.entity_category == EntityCategory.DIAGNOSTIC
    ]

    assert len(entries) == 9
    for entry in entries:
        assert entry.disabled


@pytest.mark.parametrize(
    "params",
    [
        {"entity_id": "sensor.sonic_water_flow_rate", "result": "2.137"},
        {"entity_id": "sensor.sonic_water_pressure", "result": "1910"},
        {"entity_id": "sensor.sonic_water_temperature", "result": "20"},
    ],
)
async def test_telemetry_webhook(
    hass: HomeAssistant,
    hass_client_no_auth: ClientSessionGenerator,
    mock_entry: MockConfigEntry,
    params: dict[str, str],
    mock_watergate_client: Generator[AsyncMock],
) -> None:
    """Test if water flow webhook is handled correctly."""
    await init_integration(hass, mock_entry)

    registered_entity = hass.states.get(params["entity_id"])
    assert registered_entity
    assert registered_entity.state == "unknown"

    telemetry_change_data = {
        "type": "telemetry",
        "data": {"flow": 2137, "pressure": 1910, "temperature": 20},
    }
    client = await hass_client_no_auth()
    await client.post(f"/api/webhook/{MOCK_WEBHOOK_ID}", json=telemetry_change_data)

    await hass.async_block_till_done()

    assert hass.states.get(params["entity_id"]).state == params["result"]


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
@pytest.mark.parametrize(
    "params",
    [
        {"entity_id": "sensor.sonic_ssid", "result": "Sonic"},
        {"entity_id": "sensor.sonic_rssi", "result": "-50"},
        {"entity_id": "sensor.sonic_subnet", "result": "255.255.255.0"},
        {"entity_id": "sensor.sonic_gateway", "result": "192.168.2.1"},
        {"entity_id": "sensor.sonic_ip", "result": "192.168.2.137"},
    ],
)
async def test_wifi_webhook(
    hass: HomeAssistant,
    hass_client_no_auth: ClientSessionGenerator,
    mock_entry: MockConfigEntry,
    params: dict[str, str],
    mock_watergate_client: Generator[AsyncMock],
) -> None:
    """Test if water flow webhook is handled correctly."""
    await init_integration(hass, mock_entry)

    registered_entity = hass.states.get(params["entity_id"])
    assert registered_entity
    assert registered_entity.state == "unknown"

    wifi_change_data = {
        "type": "wifi-changed",
        "data": {
            "ip": "192.168.2.137",
            "gateway": "192.168.2.1",
            "ssid": "Sonic",
            "rssi": -50,
            "subnet": "255.255.255.0",
        },
    }
    client = await hass_client_no_auth()
    await client.post(f"/api/webhook/{MOCK_WEBHOOK_ID}", json=wifi_change_data)

    await hass.async_block_till_done()

    assert hass.states.get(params["entity_id"]).state == params["result"]


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_power_supply_webhook(
    hass: HomeAssistant,
    hass_client_no_auth: ClientSessionGenerator,
    mock_entry: MockConfigEntry,
    mock_watergate_client: Generator[AsyncMock],
) -> None:
    """Test if water flow webhook is handled correctly."""
    await init_integration(hass, mock_entry)
    entity_id = "sensor.sonic_power_supply"
    registered_entity = hass.states.get(entity_id)
    assert registered_entity
    assert registered_entity.state == "unknown"

    power_supply_change_data = {
        "type": "power-supply-changed",
        "data": {"supply": "external"},
    }
    client = await hass_client_no_auth()
    await client.post(f"/api/webhook/{MOCK_WEBHOOK_ID}", json=power_supply_change_data)

    await hass.async_block_till_done()

    assert hass.states.get(entity_id).state == "external"


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_aso_report_webhook(
    hass: HomeAssistant,
    hass_client_no_auth: ClientSessionGenerator,
    mock_entry: MockConfigEntry,
    mock_watergate_client: Generator[AsyncMock],
) -> None:
    """Test if water flow webhook is handled correctly."""
    await init_integration(hass, mock_entry)
    entity_id = "sensor.sonic_last_auto_shut_off_event"
    registered_entity = hass.states.get(entity_id)
    assert registered_entity
    assert registered_entity.state == "unknown"

    valve_change_data = {
        "type": "auto-shut-off-report",
        "data": {
            "type": "VOLUME_THRESHOLD",
            "volume": 1000,
            "duration": 60,
            "timestamp": 1630000000,
        },
    }
    client = await hass_client_no_auth()
    await client.post(f"/api/webhook/{MOCK_WEBHOOK_ID}", json=valve_change_data)

    await hass.async_block_till_done()

    entity_to_check = hass.states.get(entity_id)

    assert entity_to_check.state == "2021-08-26T17:46:40+00:00"
    assert entity_to_check.attributes["volume"] == 1000
    assert entity_to_check.attributes["duration"] == 60
    assert entity_to_check.attributes["type"] == "VOLUME_THRESHOLD"
