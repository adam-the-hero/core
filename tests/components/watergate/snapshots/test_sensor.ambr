# serializer version: 1
# name: test_sensor[sensor.sonic_gateway_address-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.sonic_gateway_address',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Gateway address',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'gateway',
    'unique_id': 'a63182948ce2896a.gateway',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensor[sensor.sonic_gateway_address-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Sonic Gateway address',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_gateway_address',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_ip_address-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.sonic_ip_address',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'IP address',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'ip',
    'unique_id': 'a63182948ce2896a.ip',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensor[sensor.sonic_ip_address-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Sonic IP address',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_ip_address',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_last_auto_shut_off_event-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.sonic_last_auto_shut_off_event',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.TIMESTAMP: 'timestamp'>,
    'original_icon': None,
    'original_name': 'Last auto shut off event',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'auto_shut_off_event',
    'unique_id': 'a63182948ce2896a.auto_shut_off_event',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensor[sensor.sonic_last_auto_shut_off_event-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'timestamp',
      'friendly_name': 'Sonic Last auto shut off event',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_last_auto_shut_off_event',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_mqtt_uptime-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.sonic_mqtt_uptime',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DURATION: 'duration'>,
    'original_icon': None,
    'original_name': 'MQTT uptime',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'mqtt_uptime',
    'unique_id': 'a63182948ce2896a.mqtt_uptime',
    'unit_of_measurement': <UnitOfTime.MILLISECONDS: 'ms'>,
  })
# ---
# name: test_sensor[sensor.sonic_mqtt_uptime-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'duration',
      'friendly_name': 'Sonic MQTT uptime',
      'unit_of_measurement': <UnitOfTime.MILLISECONDS: 'ms'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_mqtt_uptime',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_power_supply-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.sonic_power_supply',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Power supply',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'power_supply',
    'unique_id': 'a63182948ce2896a.power_supply',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensor[sensor.sonic_power_supply-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Sonic Power supply',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_power_supply',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_rssi-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.sonic_rssi',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.SIGNAL_STRENGTH: 'signal_strength'>,
    'original_icon': None,
    'original_name': 'RSSI',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'rssi',
    'unique_id': 'a63182948ce2896a.rssi',
    'unit_of_measurement': 'dBm',
  })
# ---
# name: test_sensor[sensor.sonic_rssi-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'signal_strength',
      'friendly_name': 'Sonic RSSI',
      'unit_of_measurement': 'dBm',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_rssi',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_ssid-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.sonic_ssid',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'SSID',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'ssid',
    'unique_id': 'a63182948ce2896a.ssid',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensor[sensor.sonic_ssid-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Sonic SSID',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_ssid',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_subnet-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.sonic_subnet',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Subnet',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'subnet',
    'unique_id': 'a63182948ce2896a.subnet',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensor[sensor.sonic_subnet-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Sonic Subnet',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_subnet',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_uptime-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.sonic_uptime',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DURATION: 'duration'>,
    'original_icon': None,
    'original_name': 'Uptime',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'uptime',
    'unique_id': 'a63182948ce2896a.uptime',
    'unit_of_measurement': <UnitOfTime.SECONDS: 's'>,
  })
# ---
# name: test_sensor[sensor.sonic_uptime-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'duration',
      'friendly_name': 'Sonic Uptime',
      'unit_of_measurement': <UnitOfTime.SECONDS: 's'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_uptime',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_water_flow_rate-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.sonic_water_flow_rate',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.VOLUME_FLOW_RATE: 'volume_flow_rate'>,
    'original_icon': None,
    'original_name': 'Water flow rate',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_flow_rate',
    'unique_id': 'a63182948ce2896a.water_flow_rate',
    'unit_of_measurement': <UnitOfVolumeFlowRate.LITERS_PER_MINUTE: 'L/min'>,
  })
# ---
# name: test_sensor[sensor.sonic_water_flow_rate-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'volume_flow_rate',
      'friendly_name': 'Sonic Water flow rate',
      'unit_of_measurement': <UnitOfVolumeFlowRate.LITERS_PER_MINUTE: 'L/min'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_water_flow_rate',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_water_meter_duration-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.sonic_water_meter_duration',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DURATION: 'duration'>,
    'original_icon': None,
    'original_name': 'Water meter duration',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_meter_duration',
    'unique_id': 'a63182948ce2896a.water_meter_duration',
    'unit_of_measurement': <UnitOfTime.MINUTES: 'min'>,
  })
# ---
# name: test_sensor[sensor.sonic_water_meter_duration-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'duration',
      'friendly_name': 'Sonic Water meter duration',
      'unit_of_measurement': <UnitOfTime.MINUTES: 'min'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_water_meter_duration',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_water_meter_volume-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL_INCREASING: 'total_increasing'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.sonic_water_meter_volume',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.WATER: 'water'>,
    'original_icon': None,
    'original_name': 'Water Meter Volume',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_meter_volume',
    'unique_id': 'a63182948ce2896a.water_meter_volume',
    'unit_of_measurement': <UnitOfVolume.LITERS: 'L'>,
  })
# ---
# name: test_sensor[sensor.sonic_water_meter_volume-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'water',
      'friendly_name': 'Sonic Water Meter Volume',
      'state_class': <SensorStateClass.TOTAL_INCREASING: 'total_increasing'>,
      'unit_of_measurement': <UnitOfVolume.LITERS: 'L'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_water_meter_volume',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_water_pressure-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.sonic_water_pressure',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.PRESSURE: 'pressure'>,
    'original_icon': None,
    'original_name': 'Water pressure',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_pressure',
    'unique_id': 'a63182948ce2896a.water_pressure',
    'unit_of_measurement': <UnitOfPressure.MBAR: 'mbar'>,
  })
# ---
# name: test_sensor[sensor.sonic_water_pressure-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'pressure',
      'friendly_name': 'Sonic Water pressure',
      'unit_of_measurement': <UnitOfPressure.MBAR: 'mbar'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_water_pressure',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_water_temperature-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.sonic_water_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': None,
    'original_name': 'Water temperature',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'water_temperature',
    'unique_id': 'a63182948ce2896a.water_temperature',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_sensor[sensor.sonic_water_temperature-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'Sonic Water temperature',
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_water_temperature',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_sensor[sensor.sonic_wi_fi_uptime-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.sonic_wi_fi_uptime',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DURATION: 'duration'>,
    'original_icon': None,
    'original_name': 'Wi-Fi uptime',
    'platform': 'watergate',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'wifi_uptime',
    'unique_id': 'a63182948ce2896a.wifi_uptime',
    'unit_of_measurement': <UnitOfTime.MILLISECONDS: 'ms'>,
  })
# ---
# name: test_sensor[sensor.sonic_wi_fi_uptime-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'duration',
      'friendly_name': 'Sonic Wi-Fi uptime',
      'unit_of_measurement': <UnitOfTime.MILLISECONDS: 'ms'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.sonic_wi_fi_uptime',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
